src/mahoji/commands/tokkulshop.ts

```diff
@@ -91,27 +91,62 @@ export const tksCommand: OSBMahojiCommand = {
 		buy?: { name?: string; quantity?: number };
 		sell?: { name?: string; quantity?: number };
 	}>) => {
+		// Get user ID
 		const user = await mUserFetch(userID.toString());
+
+		// If user is busy don't let them go on a trip
 		if (user.minionIsBusy) return `${user.minionName} is currently busy and cannot go to the Tzhaar shops.`;
+
+		// Get user Karamja Diary completion status, item being bought, jad kc, and ironman status
 		const [hasKaramjaDiary] = await userhasDiaryTier(user, KaramjaDiary.easy);
 		const item = TokkulShopItems.find(i => stringMatches(i.name, options.buy?.name ?? options.sell?.name ?? ''));
+		const hasKilledJad: boolean = (await user.getKC(TzTokJad.id)) >= 1 ? true : false;
+		const isIronman = user.user.minion_ironman ? true : false;
+
+		// If the user is buying an invalid item
 		if (!item) return "That's not a valid item.";
-		const jadKC = await user.getKC(TzTokJad.id);
-		if (item.requireFireCape && jadKC < 1) {
-			return `You are not worthy JalYt. Before you can buy/sell ${item.name}, you need to have defeated the might TzTok-Jad!`;
+
+		// If the user is trying to buy an item that is only in Mor Ul Rek and hasn't killed Jad
+		if (item.requireFireCape && !hasKilledJad) {
+			return `You are not worthy JalYt. Before you can buy/sell ${item.name}, you need to have defeated the mighty TzTok-Jad!`;
 		}
+
+		// User bank, maxTripLength, quantity given
 		const { bank } = user;
 		const maxTripLength = calcMaxTripLength(user, activity_type_enum.TokkulShop);
 		let quantity = options.buy?.quantity ?? options.sell?.quantity ?? 1;
 		const cost = new Bank();
 		const loot = new Bank();
 
-		const amountOfRestocksNeededToBuy = item.stock ? Math.ceil(quantity / item.stock) : null;
-		const duration =
-			options.buy && amountOfRestocksNeededToBuy
-				? amountOfRestocksNeededToBuy * Time.Minute
-				: quantity * TIME_PER_1;
+		// Calculate the shop stock for the item the user wants to buy
+		let shopStock = null;
+		if (options.buy) {
+			if (hasKilledJad) {
+				if (isIronman) {
+					shopStock = item.rinIMShopStock
+						? item.rinIMShopStock
+						: item.rinShopStock
+						? item.rinShopStock
+						: item.stock
+						? item.stock
+						: null;
```

Comment:
You can simplify this into: `shopStock  = item.rinIMShopStock ?? item.rinShopStock ?? item.stock ?? null;`

Reply:
Updated, I also simplified the two other shopStock using the same format.